examples/paper_experiments/construct_AR_items.py

```diff
@@ -0,0 +1,187 @@
+    data = {}
+    label_freqs = [defaultdict(int), defaultdict(int), defaultdict(int)]
+    brand_freq = defaultdict(int)
+    for idd in raw_data:
+        if 'title' in raw_data[idd]:
+            data[idd] = raw_data[idd]
+            if idd not in asin_dict:
+                asin_dict[idd] = len(asin_dict)
+            else:
+                # cross-domain same asins
+                dup_asin.add(idd)
+            if 'brand' in data[idd] and data[idd]['brand'] != '':
+                brand_freq[data[idd]['brand']] += 1
+            if 'category' in data[idd] and len(data[idd]['category']) > 0:
+                if type(data[idd]['category'][0]) is list:
+                    breakpoint()
+                else:
+                    cate = data[idd]['category']
+                    cate = cate + [''] * (3-len(cate))
+                    cate[1] = f'{cate[0]}/{cate[1]}'
+                    cate[2] = f'{cate[1]}/{cate[2]}'
+                    label_freqs[0][cate[0]] += 1
+                    label_freqs[1][cate[1]] += 1
+                    label_freqs[2][cate[2]] += 1
+
+    for idx,layer_label in enumerate(label_freqs):
+        for k,freq in layer_label.items():
+            if freq > 1000:
+                category_dict_list[idx][k] = len(category_dict_list[idx]) + 1
+    for k,freq in brand_freq.items():
+        if freq > 1000:
+            brand_dict[k] = len(brand_dict) + 1
+        
+    
+    
+    item, item_text, brand, pt_lvl1, pt_lvl2, pt_lvl3 = [], [], [], [], [], []
+    also_buy, also_view = ([], []), ([], [])
+    for idd in data:
+        item.append(idd)
+        if 'description' in data[idd]:
+            q_text = text_process(data[idd]['title'] + ' ' + ' '.join(data[idd]['description']))
+        else:
+            q_text = text_process(data[idd]['title'])
+        item_text.append(q_text)
+        if 'brand' in data[idd] and data[idd]['brand'] != '':
+            if data[idd]['brand'] in brand_dict:
+                brand.append(brand_dict[data[idd]['brand']])
+            else:
+                brand.append(math.nan)
+        else:
+            brand.append(math.nan)
+        if 'category' in data[idd] and len(data[idd]['category']) > 0:
+            cate = data[idd]['category']
+            cate = cate + [''] * (3-len(cate))
+            cate[1] = f'{cate[0]}/{cate[1]}'
+            cate[2] = f'{cate[1]}/{cate[2]}'
+            if cate[0] in category_dict_list[0]:
+                pt_lvl1.append(category_dict_list[0][cate[0]])
+            else:
+                pt_lvl1.append(math.nan)
+            if cate[1] in category_dict_list[1]:
+                pt_lvl2.append(category_dict_list[1][cate[1]])
+            else:
+                pt_lvl2.append(math.nan)
+            if cate[2] in category_dict_list[2]:
+                pt_lvl3.append(category_dict_list[2][cate[2]])
+            else:
+                pt_lvl3.append(math.nan)
+        else:
+            pt_lvl1.append(math.nan)
```

Comment:
Why we append math.nan here?

Reply:
NAN is required by gsprocessing to ignore these labels in train/val/test split